#11884: m2m darc migration script improvements (#6816)
dotnet/arcade: Documentation/Projects/M2MRenaming/Master-to-Main-renaming-guide.md

## 1. Disable Maestro subscriptions
![Maestro enabled](images/maestro-enabled.png)

Generate json data file describing Maestro migration, review it and disable all subscriptions targeting internal and GitHub repositories:

1. Download the script [m2m-dotnet.ps1](https://raw.githubusercontent.com/dotnet/arcade/master/scripts/m2m-dotnet.ps1).
2. Generate json file which describes DARC migration (safe operation which executes only DARC read operations):
```
./m2m-dotnet.ps1 -GenerateDataFile -Repository [short repository name (e.g. dotnet/wpf)]
```
3. Verify that the file `m2m-dotnet_[timestamp].json` was generated and check that subscriptions and default channels were properly filled.

> 4. Disable DARC targeting subscriptions:

Reviewer: ```suggestion
4. Disable DARC subscriptions targeting your repository:
```

> **Note:** This uses [darc](https://github.com/dotnet/arcade/blob/master/Documentation/Darc.md) and migrates default channels and subscriptions

> 1. You can optionaly run DARC migration script in a dry run mode. This doesn't update anything, but displays DARC comands which will be executed.

Reviewer: ```suggestion
1. You can optionally run DARC migration script in a dry run mode. This doesn't update anything, but displays DARC commands which would be executed.
```

```
./m2m-dotnet.ps1 -Migrate -DataFile m2m-dotnet_[timestamp].json -DryRun
```

> 2. Run DARC migration script (it's safe to be executed repeatedly):

Reviewer: ```suggestion
2. Run DARC migration script (it is safe to execute it repeatedly):
```
